### PR TITLE
Wrapper CustomObject for `hub.dataset`

### PR DESCRIPTION
**Pull request by alberttorosyan**
Integrated `hub.dataset` -> `dict` converter into aim Custom objects.
Updated `treeutils` implementation to allow collecting raw data (without constructing `CustomObject`).
Added unit-test to cover the integration.
Mark integration unit-tests as skipped if 3rd party package is not installed (previously tests would fail)

### CODE REVIEW

**Comment by alberttorosyan** [comment]
resolves #1357

File: aim/sdk/objects/plugins/hub_dataset.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    def _tensor_meta(self, tensor: hub.Tensor):
+        meta = tensor.meta
+        return {
+            "name": tensor.key,
+            "num_samples": len(tensor),
+            "htype": tensor.htype,
+            "dtype": str(tensor.dtype) if tensor.dtype else None,
+            "compression_type": "sample_compression"
+            if meta.sample_compression
+            else ("chunk_compression" if meta.chunk_compression else None),
```
Comment (devfox-se) [inline aim/sdk/objects/plugins/hub_dataset.py:34]: this is so nested :(

File: aim/sdk/run.py [hunk 2]
```diff
@@ -351,12 +351,12 @@ def __getitem__(self, key):
 
     def get(self, key, default: Any = None, strict: bool = True):
         try:
-            return self._collect(key, strict=strict)
+            return self._collect(key, strict=strict, resolve_objects=True)
```
[the remark this answers is not in the document]
Comment (mihran113) [reply aim/sdk/run.py:354]: let's keep this approach but change `resolve_objects` default value to `False` and pas it explicitly where needed

[the remark this answers is not in the document]
Comment (alberttorosyan) [reply aim/sdk/run.py]: done ✅

**Pull request merged**